### ISSUE TITLE
8.0.1462 fails to build from source

### ISSUE THREAD

**Issue by chdiza** [macvim-dev/macvim#629]
I get the following error:
```
MacVim/MMBackend.m:3159:5: warning: 'NSConnection' is deprecated: first
      deprecated in macOS 10.13 - Use NSXPCConnection instead
      [-Wdeprecated-declarations]
    NSConnection *conn = [client connectionForProxy];
    ^
/System/Library/Frameworks/Foundation.framework/Headers/NSConnection.h:18:12: note:
      'NSConnection' has been explicitly marked deprecated here
@interface NSConnection : NSObject {
           ^
26 warnings and 1 error generated.
make[1]: *** [objects/MMBackend.o] Error 1
make: *** [first] Error 2
```

Configure flags (10.13.3):
```
    --disable-netbeans
    --disable-nls
```

**Comment by chdiza** [comment]
Whoops, forgot to paste in the error:
```
MacVim/MMBackend.m:2687:36: error: too few arguments to function call, expected
      2, have 1
                    vim_chdirfile(s);
                    ~~~~~~~~~~~~~  ^
proto/misc2.pro:85:1: note: 'vim_chdirfile' declared here
int vim_chdirfile(char_u *fname, char *trigger_autocmd);
```

**Comment by splhack** [comment]
fixed by #630
